src/dbops.cpp

```diff
@@ -50,15 +50,15 @@ fs::path rootDirectory(Database *db) {
 // are includes in the result.
 // ".ddb" files/dirs are always ignored and skipped.
 // If a directory is in the input paths, they are included regardless of includeDirs
-std::vector<fs::path> getIndexPathList(fs::path rootDirectory, const std::vector<std::string> &paths, bool includeDirs) {
+std::vector<fs::path> getIndexPathList(fs::path rootDirectory, const std::vector<std::string> &paths, bool includeDirs, bool skipParentCheck) {
```

Comment:
The point of getIndexPathList is that all paths need to be part of rootDirectory.

Why do you need to skip the assertion check?

Reply:
In the end, you're just iterating the entire folder tree right?

Just use:

```
for(auto i = fs::recursive_directory_iterator(p);
                    i != fs::recursive_directory_iterator();
                    ++i ) {
...
```

I don't think you need the logic in getIndexPathList

Reply:
Except for skipping the `.ddb` folder, which you already have a check for.

Reply:
To prevent this error:

```
    if (!skipParentCheck && !io::Path(rootDirectory).hasChildren(paths)) {
        throw FSException("Some paths are not contained within: " + rootDirectory.string() + ". Did you run ddb init?");
    }
```

Because it is called on the same folder and it's not its children. We can skip this check because it is already checked by `ddb:open`

Reply:
Fixed in https://github.com/DroneDB/DroneDB/pull/120/commits/cb521d7b8911a38623b2310640819d10197220fa